std/functional.d

```diff
@@ -1068,6 +1068,146 @@ template memoize(alias fun, uint maxSize)
     }
 }
 
+/// ditto
+template memoize(alias fun, uint maxSize, bool lru = true)
+{
+    import std.traits : ReturnType;
+    import std.typecons : Tuple;
+
+    class Node
```

Comment:
A class with dynamic allocation and linked in a doubly linked list is textbook-correct but certainly out of character for phobos and for high-performance applications. Per @MartinNowak we'd be looking at a properly managed contiguous array,